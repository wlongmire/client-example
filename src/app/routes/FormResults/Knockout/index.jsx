import React, { Component, PropTypes } from 'react';

import { LinkContainer } from 'react-router-bootstrap'

import { connect } from 'react-redux';
import { ButtonGroup, Button } from 'react-bootstrap';

function Knockout(props) {
    const emailStatusMap = {
        "LOADING":
            <div className="emailStatus">
                <img src="https://ownersedgeassets.herokuapp.com/images/main/ajax-loader.gif"/>
                <p>Emails Currently Being Processed</p><span>From there, all forms needed will be sent to argo and your inbox.</span>
            </div>,
        "ERROR":
            <div className="emailStatus error">
                <p>There appears to be something wrong.</p><span>Please contact us to complete this transaction.</span>
            </div>,
        "SUCCESS":
            <div className="emailStatus success">
                <img src="https://ownersedgeassets.herokuapp.com/images/main/thumbs-up.png"/>
                <p>Your submission forms have successfully.</p><span>Please check your your inbox. Thank you for using Argo Limited.</span>
            </div>
    }
    
    const { reason } = props.ratings[props.submission.type];
    const reasonDisplay = reason.map((r, idx)=>(
        <div key={idx} className="reason"> {r}</div>
    ))

    

    return (
    <form>
        <h3>We are reviewing your Submission.</h3>
        
        <div className="content">
            <p>Based on your answers, we couldn't provide you with an instant quote for the following reasons:</p>
            
            <div className="reasons">
                { reasonDisplay }
            </div>

            <p>One of our underwriters will be in contact with you to finalize your coverage options and assist you with purchase.</p>

            <ul>
                <li>Jessica Buelow – Supervisor – New York - 212-607-8829</li>
            </ul>
        </div>
        
        { emailStatusMap[props.emailStatus] }
        
        <ButtonGroup>
            <LinkContainer to="/submissions">
                <Button className="btn"> Return to Submissions</Button>
            </LinkContainer>
        </ButtonGroup>

    </form>
    );
}

export default connect()(Knockout)