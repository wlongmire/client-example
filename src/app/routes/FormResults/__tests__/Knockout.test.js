/* eslint-env node, jest */

import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import { Knockout } from '../Knockout'

// jest.mock('./../../../../actions/userActions', () => 'testActions')
// jest.mock('./../../../../actions/submissionActions', () => 'testActions')

const bundleRatings = {
  '13bf709d-0440-4c0a-ac26-5698afc956f2': {
    instantQuote: false,
    reason: [
      'Project cost is > than $30 million.',
      'Anticipated Project Length is > 36 months.'
    ]
  },
  'c8037b9d-0440-4c0a-ac26-5698afc956f2': {
    instantQuote: false,
    reason: [
      'Project cost is > than $30 million.',
      'Anticipated Project Length is > 36 months.'
    ]
  },
  oi: {
    instantQuote: false,
    reason: [
      'Project cost is > than $30 million.',
      'Anticipated Project Length is > 36 months.'
    ]
  }
}

const submission = {
  type: 'oi',
  clearanceStatus: 'pass'
}

// const userPropsWithBundles = {
//   bundles: [
//     {
//       basePremiumMulitplier: 0.9,
//       quoteScreenMessage: 'RPS discount, contingent on binding with Builder’s Risk Package',
//       pricingSummaryContent: 'RPS discount, contingent on binding with Builder’s Risk Package',
//       productType: 'oi',
//       bindOrderMessage: 'Is the Builders Risk bound with the RPS Package Program?',
//       id: 'c8037b9d-0440-4c0a-ac26-5698afc956f2',
//       name: 'rps'
//     },
//     {
//       basePremiumMulitplier: 1.1,
//       quoteScreenMessage: 'Test discount, contingent on binding with Builder’s Risk Package',
//       pricingSummaryContent: 'Test discount, contingent on binding with Builder’s Risk Package',
//       productType: 'oi',
//       bindOrderMessage: 'Test. Is the Interpol Risk bound with the Bumbling Package Program?',
//       id: '13bf709d-0440-4c0a-ac26-5698afc956f2',
//       name: 'interpol'
//     }
//   ],
//   subId: 'b4a94152-ef9c-40e6-b3b7-8b38fba6ab96',
//   username: 'dummyTest@gmail.com',
//   email: 'dummyTest@gmail.com',
//   broker: 'test-7fd-b3ff-4fd3-9fc2-e752b9f5b002',
//   expiration: 'dummyExpiration'
// }

// const userPropsWithoutBundles = {
//   bundles: [],
//   subId: 'b4a94152-ef9c-40e6-b3b7-8b38fba6ab96',
//   username: 'dummyTest@gmail.com',
//   email: 'dummyTest@gmail.com',
//   broker: 'test-7fd-b3ff-4fd3-9fc2-e752b9f5b002',
//   expiration: 'dummyExpiration'
// }

describe('>>> Knockout Component with Email Success --- Snapshot', () => {
  it('+++capturing Snapshot with', () => {
    const renderedValue = shallow(
      <Knockout
        submission={submission}
        emailStatus={'SUCCESS'}
        ratings={bundleRatings}
      />)
    expect(toJson(renderedValue)).toMatchSnapshot()
  })
})


describe('>>> Quote Component with Email Pending --- Snapshot', () => {
  it('+++capturing Snapshot with', () => {
    const renderedValue = shallow(
      <Knockout
        submission={submission}
        emailStatus={'PENDING'}
        ratings={bundleRatings}
      />)
    expect(toJson(renderedValue)).toMatchSnapshot()
  })
})