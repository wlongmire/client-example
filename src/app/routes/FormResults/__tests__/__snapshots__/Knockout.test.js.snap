// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`>>> Knockout Component with Email Success --- Snapshot +++capturing Snapshot with 1`] = `
<form>
  <h3>
    We are reviewing your Submission.
  </h3>
  <div
    className="content"
  >
    <p>
      Based on your answers, we couldn't provide you with a instant pricing indication for the following reasons:
    </p>
    <div
      className="reasons"
    >
      <div
        className="reason"
      >
         
        Project cost is &gt; than $30 million.
      </div>
      <div
        className="reason"
      >
         
        Anticipated Project Length is &gt; 36 months.
      </div>
    </div>
    <p>
      One of our underwriters will be in contact with you to finalize your coverage options and assist you with purchase.
    </p>
    <ul>
      <li>
        Jessica Buelow
         – 
        Supervisor
         – 
        New York
         - 
        212-607-8829
      </li>
    </ul>
  </div>
  <div
    className="emailStatus success"
  >
    <img
      alt="thubms up"
      src="https://s3.us-east-1.amazonaws.com/ownersedge-dev-assets/images/thumbs-up.png"
    />
    <p>
      Your submission forms have been successfully processed.
    </p>
    <span>
      Please check your your inbox. Thank you for using Argo Limited.
    </span>
  </div>
  <ButtonGroup
    block={false}
    bsClass="btn-group"
    justified={false}
    vertical={false}
  >
    <LinkContainer
      action="push"
      onlyActiveOnIndex={false}
      to="/submissions"
    >
      <Button
        active={false}
        block={false}
        bsClass="btn"
        bsStyle="default"
        className="btn"
        disabled={false}
      >
         Return to Submissions
      </Button>
    </LinkContainer>
  </ButtonGroup>
</form>
`;

exports[`>>> Quote Component with Email Pending --- Snapshot +++capturing Snapshot with 1`] = `
<form>
  <h3>
    We are reviewing your Submission.
  </h3>
  <div
    className="content"
  >
    <p>
      Based on your answers, we couldn't provide you with a instant pricing indication for the following reasons:
    </p>
    <div
      className="reasons"
    >
      <div
        className="reason"
      >
         
        Project cost is &gt; than $30 million.
      </div>
      <div
        className="reason"
      >
         
        Anticipated Project Length is &gt; 36 months.
      </div>
    </div>
    <p>
      One of our underwriters will be in contact with you to finalize your coverage options and assist you with purchase.
    </p>
    <ul>
      <li>
        Jessica Buelow
         – 
        Supervisor
         – 
        New York
         - 
        212-607-8829
      </li>
    </ul>
  </div>
  <ButtonGroup
    block={false}
    bsClass="btn-group"
    justified={false}
    vertical={false}
  >
    <LinkContainer
      action="push"
      onlyActiveOnIndex={false}
      to="/submissions"
    >
      <Button
        active={false}
        block={false}
        bsClass="btn"
        bsStyle="default"
        className="btn"
        disabled={false}
      >
         Return to Submissions
      </Button>
    </LinkContainer>
  </ButtonGroup>
</form>
`;
